File: examples/nodejs/src/snippets/document/analyse-multipage.ts
import * as ScanbotSDK from "scanbotsdk";
import { extractImages } from "../utils/utils";

export class AnalyzeMultiPageSnippet {
  public static async run(filePath: string): Promise<void> {
    const licenseInfo = await ScanbotSDK.getLicenseInfo();
    if (licenseInfo.status !== "OKAY") {
      console.warn("License is not valid.");
      return;
    }

    const analyzeConfig =
      new ScanbotSDK.DocumentQualityAnalyzerConfiguration();
    analyzeConfig.tileSize = 300;
    analyzeConfig.detectOrientation = true;
    analyzeConfig.minEstimatedNumberOfSymbolsForDocument = 20;
    // configure other parameters as needed

    // Extract images from the multi-page file
    const extractionResult = await extractImages(filePath);

    // `await using` ensures the analyser is properly disposed
    // when the scope ends, as it holds unmanaged resources.
    await using analyzer = await ScanbotSDK.DocumentQualityAnalyzer.create(analyzeConfig);
    const pages = extractionResult.pages;
    console.log("Pages in document: " + pages.length);

    for (let pageIndex = 0; pageIndex < pages.length; pageIndex++) {
      const page = pages[pageIndex];
      const images = page?.images;

      if (images && images.length > 0) {
        for (let imageIndex = 0; imageIndex < images.length; imageIndex++) {
          const extractedImage = images[imageIndex]?.image;
          if (extractedImage) {
            // `await using` ensures the result is properly disposed
            // when the scope ends, as it holds unmanaged resources.
            await using result = await analyzer.run(extractedImage);
            console.log(
              `Page ${pageIndex + 1}, Image ${imageIndex + 1} -> Found: ${
                result.documentFound
              }, Quality: ${result.quality}`
            );
          }
        }
      }
    }
  }
}

File: examples/python/minimal/snippets/document/analyse_multi_page.py
from scanbotsdk import *
    
def analyse_multi_page(file_path: str):
    configuration = DocumentQualityAnalyzerConfiguration(
        tile_size = 300,
        detect_orientation = True,
        min_estimated_number_of_symbols_for_document = 20
    )
    # Configure other parameters as needed.
    
    file_access_source = RandomAccessSource(file_path=file_path)
    extractor = MultiPageImageExtractor()
    result: PageExtractionResult = extractor.run(source=file_access_source)
    
    analyser = DocumentQualityAnalyzer(configuration=configuration)
    
    for page_index, page in enumerate(result.pages):
        for image_index, extracted_image in enumerate(page.images):
            result = analyser.run(image=extracted_image.image)
            print(
                f"Page {page_index + 1}, Image {image_index + 1} "
                f"-> Found: {result.document_found}, Quality: {result.quality.name}"
            )

